Mathlib/Order/Lattice.lean

```diff
@@ -1284,5 +1284,5 @@ instance [LinearOrder α] : LinearOrder (ULift.{v} α) :=
 end ULift
 
 --To avoid noncomputability poisoning from `Bool.completeBooleanAlgebra`
-instance Bool.instDistribLattice : DistribLattice Bool :=
-  inferInstance
+instance Bool.instPartialOrder : PartialOrder Bool := inferInstance
```

Comment:
Why is this new instance needed here? The reasoning mentioned here doesn't seem to apply.

Reply:
Why doesn't it seem to apply? Maybe you are unaware of the `CompleteBooleanAlgebra -> CompletePartialOrder -> PartialOrder` path?

If you're unconvinced, remove it, wait 50min for mathlib to build, and you should be able to reproduce the error

Reply:
It doesn't seem to apply because that's exactly what I did, and there was no error! :)

Reply:
Hmm, maybe the bad path isn't being hit anymore. The new instance is definitely *theoretically* necessary, so I'll move it to another PR if it's not needed here anymore